src/wallet/db.cpp

```diff
@@ -81,29 +90,38 @@ void CDBEnv::Reset()
     fMockDb = false;
 }
 
-CDBEnv::CDBEnv()
+CDBEnv::CDBEnv(const fs::path& dir_path) : strPath(dir_path.string())
 {
     Reset();
 }
 
 CDBEnv::~CDBEnv()
 {
-    EnvShutdown();
+    Close();
 }
 
 void CDBEnv::Close()
 {
+    for (auto& db : mapDb) {
```

Comment:
While you're at it, can you just merge EnvShutdown and Close? EnvShutdown is now only called from Close().